### PR TITLE
Fix auth bugs

### PR DESCRIPTION
**Pull request by sebastus**
Fixes issue #103 
Fixes issue #100

### CODE REVIEW

File: pkg/landingzone/landingzone.go [hunk 1]
```diff
@@ -144,16 +144,30 @@ func getIndentity(acct azure.Subscription, targetSubID string) azure.Identity {
...
+		for _, id := range vmIdentities.IDList {
+
+			if systemAssigned {
+				if id.DisplayName == "SystemAssigned" {
+					return id
+				}
+			} else if (userAssignedByObjectID && id.ObjectID == vmIdentityID) || (userAssignedByClientID && id.ClientID == vmIdentityID) {
+				return id
 			}
 		}
```
Comment (benc-uk) [inline pkg/landingzone/landingzone.go:179]: What if we get here and have no match? we end up falling through to 187 which is bad

**Review by benc-uk — CHANGES_REQUESTED** [review]
One comment

File: pkg/landingzone/landingzone.go (hunk 1 above)
Comment (sebastus) [reply pkg/landingzone/landingzone.go:179]: added a default case for this.

**Review by benc-uk — APPROVED** [review]

**Review by benc-uk — APPROVED** [review]
LGTM

**Pull request merged**